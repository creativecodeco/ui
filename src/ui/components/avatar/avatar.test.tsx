import { render } from '@testing-library/react';

import Avatar from './avatar.component';

describe('<Avatar />', () => {
  it('Snapshot', () => {
    global.URL.canParse = jest.fn(() => false);

    const { container } = render(<Avatar src='Test' />);

    expect(container).toMatchSnapshot();
  });

  it('Snapshot URL', () => {
    global.URL.canParse = jest.fn(() => true);

    const { container } = render(
      <Avatar src='https://daisyui.com/images/stock/photo-1534528741775-53994a69daeb.jpg' />
    );

    expect(container).toMatchSnapshot();
  });

  it('Get Letters', () => {
    global.URL.canParse = jest.fn(() => false);

    const { getByText } = render(<Avatar src='Test' />);

    expect(getByText('TE')).toBeInTheDocument();
  });

  it('Get Letters Two Names', () => {
    global.URL.canParse = jest.fn(() => false);

    const { getByText } = render(<Avatar src='Test Name' />);

    expect(getByText('TN')).toBeInTheDocument();
  });

  it('With status offline', () => {
    global.URL.canParse = jest.fn(() => false);

    const { getByTestId } = render(<Avatar src='Test Name' withStatus />);

    const avatar = getByTestId('avatar');

    expect(avatar.className.includes('offline')).toEqual(true);
  });

  it('With status online', () => {
    global.URL.canParse = jest.fn(() => false);

    const { getByTestId } = render(
      <Avatar src='Test Name' withStatus isOnline />
    );

    const avatar = getByTestId('avatar');

    expect(avatar.className.includes('online')).toEqual(true);
  });

  it('Rounded', () => {
    global.URL.canParse = jest.fn(() => false);

    const { getByTestId } = render(<Avatar src='Test Name' rounded />);

    const avatar = getByTestId('avatar-content');

    expect(avatar.className.includes('rounded-full')).toEqual(true);
  });

  it('Ring', () => {
    global.URL.canParse = jest.fn(() => false);

    const { getByTestId } = render(<Avatar src='Test Name' ring />);

    const avatar = getByTestId('avatar-content');

    expect(avatar.className.includes('avatar-ring')).toEqual(true);
  });
});
