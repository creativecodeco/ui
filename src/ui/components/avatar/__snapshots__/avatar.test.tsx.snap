// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> Snapshot 1`] = `
<div>
  <div
    class="avatar placeholder"
    data-testid="avatar"
  >
    <div
      class="bg-neutral text-neutral-content avatar-size-md rounded"
      data-testid="avatar-content"
    >
      <span>
        TE
      </span>
    </div>
  </div>
</div>
`;

exports[`<Avatar /> Snapshot URL 1`] = `
<div>
  <div
    class="avatar"
    data-testid="avatar"
  >
    <div
      class="bg-neutral text-neutral-content avatar-size-md rounded"
      data-testid="avatar-content"
    >
      <img
        alt="Avatar"
        src="https://daisyui.com/images/stock/photo-1534528741775-53994a69daeb.jpg"
      />
    </div>
  </div>
</div>
`;
